samples/nrf9160/mqtt_simple/src/main.c

```diff
@@ -332,10 +368,38 @@ static void client_init(struct mqtt_client *client)
 
 	/* MQTT transport configuration */
 #if defined(CONFIG_MQTT_LIB_TLS)
+	struct mqtt_sec_config *tls_cfg = &(client->transport).tls.config;
+	static sec_tag_t sec_tag_list[] = { CONFIG_MQTT_TLS_SEC_TAG };
+
 	client->transport.type = MQTT_TRANSPORT_SECURE;
+
+	tls_cfg->peer_verify = CONFIG_MQTT_TLS_PEER_VERIFY;
+	tls_cfg->cipher_count = 0;
+	tls_cfg->cipher_list = NULL;
+	tls_cfg->sec_tag_count = ARRAY_SIZE(sec_tag_list);
+	tls_cfg->sec_tag_list = sec_tag_list;
+	tls_cfg->hostname = CONFIG_MQTT_BROKER_HOSTNAME;
+
+#if defined(CONFIG_BSD_LIBRARY)
+	tls_cfg->session_cache = IS_ENABLED(CONFIG_MQTT_TLS_SESSION_CACHING) ?
+					    TLS_SESSION_CACHE_ENABLED :
+					    TLS_SESSION_CACHE_DISABLED;
+#else
+	/* TLS session caching is not supported by the Zephyr network stack */
+	tls_cfg->session_cache = TLS_SESSION_CACHE_DISABLED;
+
+	err = certificates_provision();
```

Comment:
If I understand correctly, the user will need to provision certificate to the modem manually, before running the sample. Perhaps we could automate the process, just as `https_client` sample does, and pre-load the certificate into the modem in the sample?
https://github.com/nrfconnect/sdk-nrf/blob/master/samples/nrf9160/https_client/src/main.c#L171
Note, that certificate should be loaded before `modem_configure` (autoconnect is already disabled in this sample).

Reply:
You are correct. Your suggestion sounds like a fine improvement. I will add certificate provisioning for the nrf91.